ambry-clustermap/src/main/java/com.github.ambry.clustermap/ClusterMapManager.java

```diff
@@ -333,10 +461,9 @@ public void addReplicas(PartitionId partitionId, String dataCenterName) {
       index++;
     }
     Datacenter datacenterToAdd = hardwareLayout.findDatacenter(dataCenterName);
-    List<Datacenter> datacentersToAdd = new ArrayList<Datacenter>();
-    datacentersToAdd.add(datacenterToAdd);
     List<Disk> disksForReplicas =
-        allocateDisksForPartition(numberOfReplicasPerDatacenter, capacityOfReplicasInBytes, datacentersToAdd);
+        allocateDisksForPartitionWithPotentialRetry(numberOfReplicasPerDatacenter, capacityOfReplicasInBytes,
```

Comment:
retry makes it sound like trying the same thing (and makes you wonder about the number of times). Perhaps call this method the same as what it was previously, and change the flag to `attemptNonRackAwareOnFailure` or something to that effect?